pkgs/development/compilers/swift/compiler/default.nix

```diff
@@ -395,17 +410,31 @@ in stdenv.mkDerivation {
     "
     buildProject llvm llvm-project/llvm
 
+    ${lib.optionalString stdenv.isDarwin ''
+    # Add appleSwiftCore to the search paths. We can't simply add it to
+    # buildInputs, because it is potentially an older stdlib than the one we're
+    # building. We have to remove it again after the main Swift build, or later
+    # build steps may fail. (Specific case: Concurrency backdeploy uses the
+    # Sendable protocol, which appears to not be present in the macOS 11 SDK.)
+    OLD_NIX_SWIFTFLAGS_COMPILE="$NIX_SWIFTFLAGS_COMPILE"
+    OLD_NIX_LDFLAGS="$NIX_LDFLAGS"
+    export NIX_SWIFTFLAGS_COMPILE+=" -I ${appleSwiftCore}/lib/swift"
+    export NIX_LDFLAGS+=" -L ${appleSwiftCore}/lib/swift"
+    ''}
```

Comment:
```suggestion
  '' + lib.optionalString stdenv.isDarwin ''
    # Add appleSwiftCore to the search paths. We can't simply add it to
    # buildInputs, because it is potentially an older stdlib than the one we're
    # building. We have to remove it again after the main Swift build, or later
    # build steps may fail. (Specific case: Concurrency backdeploy uses the
    # Sendable protocol, which appears to not be present in the macOS 11 SDK.)
    OLD_NIX_SWIFTFLAGS_COMPILE="$NIX_SWIFTFLAGS_COMPILE"
    OLD_NIX_LDFLAGS="$NIX_LDFLAGS"
    export NIX_SWIFTFLAGS_COMPILE+=" -I ${appleSwiftCore}/lib/swift"
    export NIX_LDFLAGS+=" -L ${appleSwiftCore}/lib/swift"
  '' + ''
```